File: ruby/gmsec4/dox/MessageSpecification.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.


module Libgmsec_ruby #:nodoc: don't document this

	##
	# This class represents a message definition derived from a message schema.
	#
	# ==== See also
	#
	# Libgmsec_ruby::FieldSpecification
	#
	# Libgmsec_ruby::Specification.get_message_specifications
	#
	class MessageSpecification
		##
		# Returns the schema (template) ID for this Message Specification.
		#
		# ==== Returns
		#
		# The schema ID as a string.
		#
		def get_schema_id
		end


		##
		# Returns an array of FieldSpecification objects for this Message Specification.
		#
		# ==== Returns
		#
		# An array of Libgmsec_ruby::FieldSpecification objects.
		#
		def get_field_specifications
		end
	end
end

File: nodejs/gmsec5/dox/MessageSpecification.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file MessageSpecification.java
 */

package gmsec;


/**
 * The MessageSpecification class represents a message defintion derived
 * from a message schema.
 *  
 * @see FieldSpecification
 * @see Specification
 * @see MessageFactory.getSpecification()
 */
public class MessageSpecification
{
	/**
	 * Returns the schema (template) ID for this Message Specification.
	 *
	 * @return The schema ID for this Message Specification.
	 */
	public getSchemaID();


	/**
	 * Returns the subject template for this Message Specification.
	 *
	 * @return The subject template for this Message Specification.
	 */
	public getSubjectTemplate();


	/**
	 * Returns a FieldSpecificationArray of FieldSpecification objects for this Message Specification.
	 *
	 * @return An array of FieldSpecification objects.
	 */
	public getFieldSpecifications();
}
